examples/encodevideo.jl

```diff
@@ -0,0 +1,160 @@
+# Based on https://www.ffmpeg.org/doxygen/trunk/encode_video_8c-example.html
+using VideoIO, Printf
+
+"""
+encode(enc_ctx::Ptr{VideoIO.AVCodecContext}, frame, pktptr::Ptr{VideoIO.AVPacket}, io::IO)
+
+Append and encode frame to output.
+"""
+function encode(enc_ctx::Ptr{VideoIO.AVCodecContext},
+    frame, pktptr::Ptr{VideoIO.AVPacket}, io::IO)
+    ret = VideoIO.avcodec_send_frame(enc_ctx, frame)
+    if ret < 0
+        error("Error $ret sending a frame for encoding")
+    end
+
+    while ret >= 0
+        ret = VideoIO.avcodec_receive_packet(enc_ctx, pktptr)
+        if (ret == -35 || ret == -541478725) # -35=EAGAIN -541478725=AVERROR_EOF
+             return
+        elseif (ret < 0)
+            error("Error $ret during encoding")
+        end
+        pkt = unsafe_load(pktptr)
+        println("Write packet $(pkt.pts) (size=$(pkt.size))")
+        data = unsafe_wrap(Array,pkt.data,pkt.size)
+        write(io,data)
+        VideoIO.av_packet_unref(pktptr)
+    end
+end
+
+filename = "video.h264"
+codec_name = "libx264"
+
+c = VideoIO.AVCodecContext[]
+
+endcode = UInt8[0, 0, 1, 0xb7]
+
+codec = VideoIO.avcodec_find_encoder_by_name(codec_name)
+if codec == C_NULL
+    error("Codec '$codec_name' not found")
+end
+
+c = VideoIO.avcodec_alloc_context3(codec)
+#c = Ptr{VideoIO.AVCodecContext}[VideoIO.avcodec_alloc_context3(codec)]
```

Comment:
The commented-out code is necessary so you can pass `c` to `avcodec_free_context` below.